### PR TITLE
Feature/fun call

### PR DESCRIPTION
**Pull request by hhssman**
This feature will allow the use of function calls in expressions.

Before you had to give it a variable to store the result in before it could be used in a expression
Now you can call a service in a expression where you want to use the result.

Example: 
`result = operation@outputPort(request) + 1`

### CODE REVIEW

File: jolie/src/main/java/jolie/runtime/expression/SolicitResponseExpression.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	@Override
+	public Expression cloneExpression( TransformationReason reason ) {
+		return new SolicitResponseExpression( operationId, outputPort, outputExpression, types );
+	}
+
+	@Override
+	public Value evaluate() throws FaultException {
+		// TODO implement all exception cases
```
[the remark this answers is not in the document]
Comment (hhssman) [reply jolie/src/main/java/jolie/runtime/expression/SolicitResponseExpression.java:45]: Yes that is resolved I just forgot to remove it again

File: jolie/src/main/java/jolie/runtime/VariablePath.java [hunk 2]
```diff
@@ -87,22 +87,26 @@ public final VariablePath containedSubPath( VariablePath otherVarPath ) {
...
+				expr = pair.value();
+				otherExpr = otherPair.value();
+
+				myIndex = (expr == null) ? 0 : expr.evaluate().intValue();
+				otherIndex = (otherExpr == null) ? 0 : otherExpr.evaluate().intValue();
+				if( myIndex != otherIndex )
+					return null;
+			}
+		} catch( FaultException e ) {
+			throw new AssertionError( "It should mot ne possible that a FaultExeption was thrown here", e.getCause() );
```
[the remark this answers is not in the document]
Comment (hhssman) [reply jolie/src/main/java/jolie/runtime/VariablePath.java:109]: I am not entirely sure what you mean with that. Do you mean that you can do something like this `x.( call@O(y) ) = 5`? If so I did not realize that was a possibility and it should instead throw an exception.

[the remark this answers is not in the document]
Comment (fmontesi) [reply jolie/src/main/java/jolie/runtime/VariablePath.java:109]: Yes that's what I mean. :) It looks feasible now.

File: libjolie/src/main/java/jolie/lang/parse/OLParser.java [hunk 3]
```diff
@@ -3329,6 +3331,13 @@ private OLSyntaxNode parseFactor()
 
 		if( path != null ) {
 			switch( token.type() ) {
+			case AT:
+				nextToken();
```
Comment (fmontesi) [inline libjolie/src/main/java/jolie/lang/parse/OLParser.java:3335]: Mmmh, what happens if one writes `x.y@Hello( "x" )`, which should be forbidden? It looks like you'll parse it incorrectly as x@Hello("x").

**Pull request closed without merging**